### ISSUE TITLE
Hinted Handoff & Harvest [JIRA: RIAK-1709]

### ISSUE THREAD

**Issue by rzezeski** [basho/yokozuna#125]
tl;dr: If data is written while nodes are down then when those nodes come back up there will be a window where Yokozuna gives < 100% harvest on queries involving that data.

`harvest = data available / complete data`

While nodes are down data will be written to secondary vnodes.  This is known as sloppy quorum.  Also, Yokozuna will take the downed nodes out of query rotation.

When the primary vnodes come back up Yokozuna will put those nodes back in query rotation.  In almost all cases this will happen BEFORE that secondary vnodes have handed the new data to the primaries via hinted handoff.  If Yokozuna decides to use those nodes/vnodes for a query then it will have degraded harvest.  But there is no reason it has to be this way because there are other replicas available.

Solution: Yokozuna coverage needs to take hinted handoff into account.  If a primary vnode/partition is waiting for data from a fallback then it must be taken out of the query rotation.

**Comment by rzezeski** [comment]
Also, this is not just an issue for Yokozuna.  Anything in Riak
relying on riak_core coverage has this problem.  That includes things
like list keys and 2i.  Coverage needs to be able to take hinted
handoff into account and perhaps have harvest vs yield knobs.

**Comment by DSomogyi** [comment]
Comment for Jira.
